pulp_ansible/app/models.py

```diff
@@ -291,6 +313,278 @@ class Meta:
+        if self.rekor_url == self.PUBLIC_REKOR_URL:
+            return RekorClient.production(self.trust_updater)
+        if self.rekor_url == self.STAGING_REKOR_URL:
+            return RekorClient.staging(self.trust_updater)
+
+        rekor_key = RekorKeyring(Keyring(self.rekor_public_keys))
+        ctfe_key = CTKeyring(Keyring())
+        return RekorClient(
+            self.rekor_url,
+            rekor_key,
+            ctfe_key,
+        )
+
+    @property
+    def certificates_chain(self):
+        """Get the Fulcio certificate chain."""
+        return load_pem_x509_certificates(bytes(self.certificate_chain, "utf-8"))
+
+    @property
+    def verifier(self):
+        """Get a Verifier instance."""
+        if self.rekor_url == self.PUBLIC_REKOR_URL:
+            return Verifier.production()
+        if self.rekor_url == self.STAGING_REKOR_URL:
+            return Verifier.staging()
+
+        return Verifier(
+            rekor=self.rekor,
+            fulcio_certificate_chain=self.certificates_chain,
+        )
+
+    def sigstore_verify(self, manifest, sigstore_bundle):
+        """Verify a Sigstore signature validity."""
+        verification_materials = VerificationMaterials.from_bundle(
+            input_=manifest, bundle=sigstore_bundle, offline=self.verify_offline
+        )
+
+        policy = Identity(
+            identity=self.expected_identity,
+            issuer=self.expected_oidc_issuer,
+        )
+        return self.verifier.verify(materials=verification_materials, policy=policy)
+
+    @hook(BEFORE_SAVE)
+    def format_pem_pubkeys(self):
+        """Override the base `save` method to properly format PEM-encoded files."""
+        if self.rekor_root_pubkey:
+            self.rekor_root_pubkey = validate_and_format_pem_public_key(self.rekor_root_pubkey)
+
+    class Meta:
+        default_related_name = "%(app_label)s_%(model_name)s"
+
+
+class CollectionVersionSigstoreSignature(Content):
+    """
+    A content type representing a Sigstore signature attached to a content unit.
+    Fields:
+        sigstore_bundle (models.JSONField):
+            An optional Sigstore bundle used for offline verification.
+        sigstore_bundle_sha256_hash (CharField):
+            A sha256 hash value for the Sigstore bundle.
+            Fixes error related to index row size exceeding btree version 4
+            maximum 2704 for indexes when creating unique_together constraint.
+            See https://code.djangoproject.com/ticket/14904#no1
+    Relations:
+        signed_collection (models.ForeignKey):
+            A collection version this signature is relevant to.
+        sigstore_signing_service (models.ForeignKey):
+            The Sigstore Siging Service used for signing the collection version.
+    """
+
+    TYPE = "collection_sigstore_signatures"
+
+    sigstore_bundle = models.JSONField(null=True)
+    sigstore_bundle_sha256_hash = models.CharField(max_length=64, unique=True, db_index=True, null=True)
```

Comment:
```suggestion
    sigstore_bundle_sha256_hash = models.CharField(max_length=64, db_index=True, null=True)
```
`unique_together` is enough. Don't forget to remove the same code from the migration file.